### PR TITLE
i18n: Merge duplicate messages with different IDs

### PR DESCRIPTION
**Pull request by Betree**
Based on a suggestion from @leonardohof, this PR adds a warning on `npm run build:langs` when finding duplicate messages with different ids. Example:
```jsx
<FormattedMessage id="loading" defaultMessage="Loading" />
<FormattedMessage id="expenses.loading" defaultMessage="Loading" />
```

> npm run build:langs
> ...
> :information_source: Found similar message for IDs (startDateAndTime, startDateAndTimeBis): start date and time
> :heavy_check_mark: Done!

I've also merged the existing duplicates (we had around 160+ duplicated strings).

### CODE REVIEW

File: scripts/translate.js [hunk 1]
```diff
@@ -97,6 +103,21 @@ const translate = (locale, defaultMessages, updatedKeys) => {
   fs.writeFileSync(`${LANG_DIR}${locale}.json`, translations);
 };
 
+const warnForDuplicateMessages = messages => {
+  const ignoredMessages = new Set(['all', 'type', 'paid', 'pending', 'other']);
```
Comment (Betree) [inline scripts/translate.js]: Exceptions can be configured here.

Comment (kewitz) [reply scripts/translate.js]: 💯

File: scripts/translate.js [hunk 2]
```diff
@@ -97,6 +103,21 @@ const translate = (locale, defaultMessages, updatedKeys) => {
...
+  const ignoredIds = new Set(['section.team.title', 'contribute.step.details']);
```
Comment (kewitz) [inline scripts/translate.js]: Perhaps put these sets on the top of the file as constants:
```js
const DUPLICATED_IGNORED_IDS
const DUPLICATED_IGNORED_KEYWORDS
```

File: scripts/translate.js [hunk 3]
```diff
@@ -115,3 +136,4 @@ supportedLocales.forEach(locale => {
 
 // Write root file at the end to only save if translations don't crash
 fs.writeFileSync(DEFAULT_TRANSLATIONS_FILE, convertToJSON(defaultMessages));
+console.info('✔️  Done!');
```
Comment (kewitz) [inline scripts/translate.js]: Cool.

**Pull request merged**